Fix behavior of color picker on sensor blocks

ivanixgames opened the pull request:
### Resolves
https://github.com/LLK/scratch-blocks/issues/1381

Expectation:
Users expect behavior to mimic that of color picker found in paint editor.
Upon tapping or clicking in the color range of the slider, the thumb should move to that position

Actual Behavior, 2 issues:
1) For hue and brightness the thumb moves incrementally while the mouse button is held down.
2) The saturation does not work at all and can only be adjusted by dragging the thumb.

### Proposed Changes
For issue 1:
a)  All 3 sliders ( Color, Saturation, Brightness ) should call setMoveToPointEnabled(true) .

For issue 2:
a)  orginal code was using range of  0 to 1 using step of 0.001.  However the sliderbase class rounds up steps so can only use steps of 1 or greater, this requires a max range to be scaled up as well, which in turn requires an external conversion (SATURATION_CONV) to be used to convert back to range of 0 to 1 outside the slider.  Note that this issue and solution is not enabled when setMoveToPointEnable(true) is called. 
However this fix should remain in case in future it is decided that incrementally thumb movement is desired again.
 
### Reason for Changes
Provide consistent color pick behavior with that on paint editor and as reported in
original issue on scratch-gui.


### Test Coverage
Test manually using Chromebrowser on test file
file://.../scratch-blocks/tests/vertical_playground.html

1) drag 'touching color' block from sensors category onto workspace.
2) click on color,  color picker dialog pop ups.
3) move mouse pointer to some part of colored range bar for each of Color, Saturation, Brightness
4) Click or press down on left mouse button.

Expectation:
-  thumb to move to position
- value to also be adjusted base on new thumb position
Results 
- passed with minor issue

Minior issue:
Repeated clicking in color range will cause thumb to move up and off of slider!
I traced this down to an issue with google ui sliderbase and its animation logic.
Issue will be submitted to github on that repo shortly later.
The issue is revealed because scratch uses thumb style property of margin-top: -1.
However, I believe this is a bug in google ui sliderbase.js, not in scratch code.

ivanixgames committed:
Fix for issue 1381
- Move thumb to position of mouse click or tap
- Fix saturation slider which was trying to use range of 0 to 1 and
  steps in fractions of whole number.
  google closure slider (sliderbase) can only handle integer values.

ivanixgames committed:
- 2nd fix for issue 1381
-  call method setMoveToPointEnabled instead of changing (private) property directlly

ivanixgames commented:
Issue and PR posted regarding minor issue mentioned above.
https://github.com/google/closure-library/issues/876

thisandagain commented:
Thanks @ivanixgames! We'll take a look at this soon. Sorry for the slow response.

paulkaplan reviewed (changes requested):
Awesome find with the `setMoveToPointEnabled`, I'm glad that is already built in. 

But can you explain why the saturation conversion is needed? Is it needed for the `setMoveToPointEnabled` change?

I tried just making the `setMoveToPointEnabled` change locally without the saturation fix and it seems to work fine?

![click-move](https://user-images.githubusercontent.com/654102/37096284-2a9a920c-21e6-11e8-92d4-da21b3db42bd.gif)
 
That is with this change
![image](https://user-images.githubusercontent.com/654102/37096292-306d11b4-21e6-11e8-9cb8-d6f1ff5529aa.png)

Also for the drifting problem, is there a way in our CSS that we can fix that? I wouldn't want to merge this without that being addressed.

ivanixgames commented:
@paulkaplan 
The saturation conversion is not needed if setMoveToPointEnabled is used.  
That patch is only if the original intended behavior was desired.

For the drifting issue. if you want to fix in css,  make sure margin-top is not set or set to 0px  for horizontal slider, and margin-left is left unset or set to 0px for vertical slider. 
See https://user-images.githubusercontent.com/858678/36078378-4a4e5a0e-0fb0-11e8-8abb-ae6ef686261d.png

paulkaplan commented:
@ivanixgames cool! Would you mind removing the saturation thing then, because we do want to use setMoveToPointEnabled?

paulkaplan commented:
@ivanixgames cool! Would you mind removing the saturation thing then, because we do want to use setMoveToPointEnabled?

paulkaplan commented:
Also @ivanixgames I found that if you just change `margin-top: -1px` to `top: -1px` (since it is position absolute) that the sliders don't drift but are still positioned correctly. So if you could make those two changes (removing the saturation change and changing margin-top to top in css.js line 837) we can merge this PR. Thanks for tracking down this issue!

ivanixgames commented:
@paulkaplan   Not mind at all, just  give me about a day to make and retest the changes.  Cheers

ivanixgames committed:
- 3rd patch for issue 1381
  - Removed saturation conversion patch
  - setMoveToPointEnabled patch left in place
  - Converted css property margin-top to top in .goog-slider-horizontal .goog-slider-thumb
